Source/Laser/LaserEnvelope.H

```diff
@@ -0,0 +1,13 @@
+#include <complex>
+#include <iostream>
+#include <complex>
+#include <vector>
+
+
+#include <iostream>
+#include <vector>
+
+class LaserEnvelope {
```

Comment:
```suggestion
class LaserEnvelope {
public:
       LaserEnvelope::AllocateLaserEnvelope (int lev, const amrex::BoxArray& ba, const amrex::DistributionMapping& dm, const amrex::IntVect& ngA, const bool aux_is_nodal)

```

Reply:
As discussed offline, here maybe we can have member functions similar to the ones of the class `HybridPICModel`: https://github.com/ECP-WarpX/WarpX/blob/de8e9622f39bec29369c37907a6d854fc619aacf/Source/FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.H#L37-L42.